### PR TITLE
Fix issue when after creating a wallet, the app is showing the onboarding screen for a while without any loading indicator

### PR DESCRIPTION
**Pull request by osdnk**
Fixes RNBW-3939

## What changed (plus any additional context for devs)
Look like on android the interactions might not happen so smoothly. I delayed hiding the loading screen. Also, the `replace` action is not working properly, so I added `navigate`, but then I remove the screen below with the custom action I implemented.

Before: [streamable.com/0nfv09](https://streamable.com/0nfv09)
After: https://streamable.com/v73zxo 
Ignore the wrong screen being focused. This is addressed in a different PR.

 
## What to test
1. Import the wallet via seeds, ens, create a new one (should have no impact on new wallet creation)
2. Test broke ens, seeds etc
3. Ignore the wrong screen being focused. This is addressed in a different PR.
4. Test no impact on iOS

### CODE REVIEW

File: src/model/wallet.ts [hunk 1]
```diff
@@ -887,7 +887,9 @@ export const createWallet = async (
           ? (walletResult as Wallet)
           : new Wallet(pkey);
       setTimeout(() => {
-        dispatch(setIsWalletLoading(null));
+        if (ios || !isImported) {
```
Comment (osdnk) [inline src/model/wallet.ts:890]: !imported = new wallet - then we use this logic for dismissing the loading state

Comment (tchayen) [reply src/model/wallet.ts:890]: Maybe it makes sense to add it as a comment in the code?

**Review by tchayen — APPROVED** [review]
Looks ok

File: src/hooks/useImportingWallet.js [hunk 2]
```diff
@@ -261,18 +267,28 @@ export default function useImportingWallet({ showImportModal = true } = {}) {
...
                     navigate(Routes.WALLET_SCREEN, { initialized: true });
                   }
+                  if (android) {
+                    handleSetImporting(false);
+                    InteractionManager.runAfterInteractions(() =>
+                      setIsWalletLoading(null)
+                    );
+                    dispatchNavigation({
+                      type: '@RAINBOW/REMOVE_FIRST',
+                    });
```
Comment (terrysahaidak) [inline src/hooks/useImportingWallet.js:290]: Can we move this entire thing to Navigation? I guess this is the place it should be

**Review by terrysahaidak — APPROVED** [review]
Looks good. One minor comment.

Also i would really want to see some more comments directly in the code since it's a bit complicated and kinda confusing why we need all these extra `if (android) {}` things. Will dramatically help future readers of this code to understand the reasons behind all of these changes.

**Review by nickbytes — APPROVED** [review]

**Review by estebanmino — APPROVED** [review]
code looks good and proof of dev qa http://recordit.co/4ayHBlCp3o looking good

**Review by brunobar79 — APPROVED** [review]

**Pull request merged**